### DISCUSSION TITLE
Firewall Bypass

### DISCUSSION THREAD

**Discussion by pwn0r** [tnodir/fort#245] [General]
I opened this as a discussion as this is mostly windows architecture fault. In any way, this affects the built-in Windows Firewall per se.
However it does apply to fort firewall either. 

It is possible sometime in the future to prevent such bypass? Or too much work? Probably would need your own kernel driver :P

Anyway, as a simple example of such bypass we can see in Oracle VirtualBox. Basically it seems that any application that is able to install their own drivers (signed of course, in this case both by oracle and MS hardware signing) would easily bypass any application firewall on the same OS?

**Comment by tnodir** [comment]
> windows architecture fault

I don't think, that is a Windows' fault.

E.g. I use VMWare virtual machine and its bridged network. Which works via separate driver. And you can not filter it by Fort.

Hm.. If you don't want a bypass, then just don't install products with drivers?
